### PR TITLE
unit tests: don't crash if !HAVE_SQLITE3.

### PR DESCRIPTION
**Pull request by rustyrussell**
Fixes: #4928
Reported-by: @whitslack
Changelog-None

### CODE REVIEW

File: configure [hunk 1]
```diff
@@ -317,7 +317,7 @@ desc=sqlite3_expanded_sql
 style=DEFINES_EVERYTHING|EXECUTE|MAY_NOT_COMPILE
 link=$SQLITE3_LDLIBS
 code=
-#include <sqlite3.h>
+#include <sqlite3x.h>
```
Comment (whitslack) [inline configure:320]: @rustyrussell: What package provides `<sqlite3x.h>`? Is C-Lightning planning to switch to a fork of SQLite? The SQLite package on my system only provides `<sqlite3.h>` and `<sqlite3ext.h>`. Was the latter renamed at some point upstream? I don't find this header included in any source files, just in these modified `configure` tests.

Comment (rustyrussell) [reply configure:320]: Oops, this is my testing hack!!

**Comment by cdecker** [comment]
ACK https://github.com/ElementsProject/lightning/pull/5140/commits/5e6e9068cbea32e9c99fcd47453114f8d7507bb0

**Comment by rustyrussell** [comment]
Trivial rebase on master.

**Pull request merged**